packages/react-debug-tools/src/__tests__/ReactHooksInspectionIntegration-test.js

```diff
@@ -434,7 +441,7 @@ describe('ReactHooksInspectionIntegration', () => {
           "isStateEditable": false,
           "name": "Memo",
           "subHooks": [],
-          "value": "Ab",
+          "value": "AB",
```

Comment:
Here and L762 the assertion is updated. 

It seems that in legacy root the useState setter completed (a=>A) but the useReducer setter did not (b=>B). I'm not sure why. I would expect the value to be AB here as it is now.

Reply:
Yeah it's the batching batching. 

It was only `Ab` before because they weren't batched, and if you look at the `useMemo` (which this snapshot is showing the state of):

```
React.useMemo(() => state1 + state2, [state1]);
```

Before, this would re-run the memo when state1 changed, which was a different render then when state 2 changed. So the memo state would only see the `state1` change. With batching, both states are updated at the same time, so even though the memo still only fires on `state1` changing, by that point `state2` has also already changed.